components/supervisor/frontend/src/ide/heart-beat.ts

```diff
@@ -70,6 +63,22 @@ export function schedule(wsInfo: WorkspaceInfo, sessionId: string): void {
         sendHeartBeat();
     }, activityInterval);
     toCancel.push(Disposable.create(() => clearInterval(intervalHandle)));
+
+    const handleUnload = async () => {
+        if (isSaaS) {
```

Comment:
For self-hosting we should always use a new approach? Only backward compatibility for SaaS?